### DISCUSSION TITLE
Outline VPN Server Speed

### DISCUSSION THREAD

**Discussion by JoZombie8** [community/community#31327] [Other Feature Feedback, Questions, & Ideas]
Hi All, 

Sorry I couldn't find somewhere else to post this. I would appreciate if you guys can help me here. 

I have a setup of Outline VPN through DigitalOcean Server (Located in USA) and I'm connecting my devices through Shadowsocks. Everything is working perfectly 

I'm just looking to increase the connection speed, as I'm noticing a huge drop in speed when connecting to the vpn. Will it be a good option to increase the size of the server adding more memory, CPU and so on ?

**Comment by msuliq** [comment]
Hi, I would suggest looking into ping between your laptop and the Digital Ocean server that you have leased.
Let's say that ping (or speed of packages reaching addressee) from your laptop directly to GitHub without vpn is 500ms.
Then we have ping from your laptop to the Digital Ocean service, which let's say is 600ms. 
Then we have ping between Digital Ocean server and GitHub server, which would be 100ms (or less since both may be located very close).
So, in case of connection without vpn, your packets reach Github in 500ms (direct connection).
In case of vpn, they reach in 700ms (600ms from your laptop to Digital Ocean VPS + 100ms from Digital Ocean to GitHub).
Adding more compute power or RAM will help if the issue is related to how fast your VPS handles the operation of vpn. Based on my personal experience, Outline usually does not require sufficient resources if run for personal use.
Looking into ping will have more impact on the speed of your connection.
For example, you could switch your Digital Ocean server to one in the Netherlands (if you are in Europe) and you will get much lower ping from your laptop to your VPS.
Hope it helps :)
